Simplify description and some rewrites (#2281)
uxlfoundation/oneDPL: documentation/library_guide/parallel_api/pass_data_algorithms.rst

Use |onedpl_short| Iterators
----------------------------

You can use iterators defined in ``<oneapi/dpl/iterator>`` header in the ``oneapi::dpl`` namespace:

- ``counting_iterator``
- ``zip_iterator``
- ``transform_iterator``
- ``discard_iterator``
- ``permutation_iterator``

:ref:`Iterators <iterator-details>` section describes them in detail and provides usage examples.

> If you want to use these iterators on top of your own custom iterators,

Author: ```suggestion
If you want to use these iterators in combination with your own custom iterators,
```

      strided_iterator operator+(difference_type n) const { return strided_iterator(ptr + n * stride, stride);}
      strided_iterator operator-(difference_type n) const { return strided_iterator(ptr - n * stride, stride); }
      difference_type operator-(const strided_iterator& other) const {return (ptr - other.ptr) / stride; }

      bool operator==(const strided_iterator& other) const { return ptr == other.ptr; }
      bool operator!=(const strided_iterator& other) const { return ptr != other.ptr; }
      bool operator<(const strided_iterator& other) const { return ptr < other.ptr; }
      bool operator>(const strided_iterator& other) const { return ptr > other.ptr; }
      bool operator<=(const strided_iterator& other) const { return ptr <= other.ptr; }
      bool operator>=(const strided_iterator& other) const { return ptr >= other.ptr; }

      // Another way to make this iterator indirectly device accessible
      // friend std::true_type is_onedpl_indirectly_device_accessible(strided_iterator) { return {}; }
  private:
      It ptr;
      difference_type stride;
  };

  // Make strided_iterator indirectly device accessible
  template <typename It>

>   std::true_type is_onedpl_indirectly_device_accessible(strided_iterator<It>);

Author: ```suggestion
  auto is_onedpl_indirectly_device_accessible(strided_iterator<It>) -> oneapi::dpl::is_indirectly_device_accessible<It>;
```
